linera-chain/src/manager.rs

```diff
@@ -99,85 +106,93 @@ pub enum Outcome {
     /// Highest-round authenticated block that we have received and checked. If there are multiple
     /// proposals in the same round, this contains only the first one.
-    #[debug(skip_if = Option::is_none)]
-    pub proposed: Option<BlockProposal>,
+    #[graphql(skip)]
+    pub proposed: RegisterView<C, Option<BlockProposal>>,
     /// Latest validated proposal that we have voted to confirm (or would have, if we are not a
     /// validator).
-    #[debug(skip_if = Option::is_none)]
-    pub locked: Option<ValidatedBlockCertificate>,
+    #[graphql(skip)]
+    pub locked: RegisterView<C, Option<ValidatedBlockCertificate>>,
     /// These are blobs published or read by the locked block.
-    #[debug(skip_if = BTreeMap::is_empty)]
-    pub locked_blobs: BTreeMap<BlobId, Blob>,
+    pub locked_blobs: MapView<C, BlobId, Blob>,
     /// Latest leader timeout certificate we have received.
-    #[debug(skip_if = Option::is_none)]
-    pub timeout: Option<TimeoutCertificate>,
+    #[graphql(skip)]
+    pub timeout: RegisterView<C, Option<TimeoutCertificate>>,
     /// Latest vote we cast to confirm a block.
-    #[debug(skip_if = Option::is_none)]
-    pub confirmed_vote: Option<Vote<ConfirmedBlock>>,
+    #[graphql(skip)]
+    pub confirmed_vote: RegisterView<C, Option<Vote<ConfirmedBlock>>>,
     /// Latest vote we cast to validate a block.
-    #[debug(skip_if = Option::is_none)]
-    pub validated_vote: Option<Vote<ValidatedBlock>>,
+    #[graphql(skip)]
+    pub validated_vote: RegisterView<C, Option<Vote<ValidatedBlock>>>,
     /// Latest timeout vote we cast.
-    #[debug(skip_if = Option::is_none)]
-    pub timeout_vote: Option<Vote<Timeout>>,
+    #[graphql(skip)]
+    pub timeout_vote: RegisterView<C, Option<Vote<Timeout>>>,
     /// Fallback vote we cast.
-    #[debug(skip_if = Option::is_none)]
-    pub fallback_vote: Option<Vote<Timeout>>,
+    #[graphql(skip)]
+    pub fallback_vote: RegisterView<C, Option<Vote<Timeout>>>,
     /// The time after which we are ready to sign a timeout certificate for the current round.
-    #[debug(skip_if = Option::is_none)]
-    pub round_timeout: Option<Timestamp>,
+    pub round_timeout: RegisterView<C, Option<Timestamp>>,
     /// The lowest round where we can still vote to validate or confirm a block. This is
     /// the round to which the timeout applies.
     ///
     /// Having a leader timeout certificate in any given round causes the next one to become
     /// current. Seeing a validated block certificate or a valid proposal in any round causes that
     /// round to become current, unless a higher one already is.
-    pub current_round: Round,
+    #[graphql(skip)]
+    pub current_round: RegisterView<C, Round>,
     /// The owners that take over in fallback mode.
-    #[debug(skip_if = BTreeMap::is_empty)]
-    pub fallback_owners: BTreeMap<Owner, (PublicKey, u64)>,
+    pub fallback_owners: RegisterView<C, BTreeMap<Owner, (PublicKey, u64)>>,
```

Comment:
That's a lot of mechanical work. I know it's not for this PR but it'd be nice if `derive(View)` wrapped all of the fields into propery `*View` variants.

Reply:
I don't know if that's possible in general. E.g. some but not all maps should turn into `MapView`s.

Reply:
sensible deafults + overrides with attributes.